src/main/java/org/frawa/elmtest/core/ElmTestJsonProcessor.java

```diff
@@ -0,0 +1,243 @@
+package org.frawa.elmtest.core;
+
+import com.google.gson.*;
+import com.intellij.execution.testframework.sm.runner.events.*;
+import org.frawa.elmtest.core.json.CompileErrors;
+import org.frawa.elmtest.core.json.Error;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static org.frawa.elmtest.core.LabelUtils.*;
+
+public class ElmTestJsonProcessor {
+
+    private static Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    private Path currentPath = LabelUtils.EMPTY_PATH;
+
+    public List<TreeNodeEvent> accept(String text) {
+        try {
+            JsonObject obj = gson.fromJson(text, JsonObject.class);
+            if (obj == null) {
+                return null;
+            }
+            JsonElement type = obj.get("type");
+            if (type != null && "compile-errors".equals(type.getAsString())) {
+                return accept(toCompileErrors(obj));
+            }
+            String event = obj.get("event").getAsString();
+
+            if ("runStart".equals(event)) {
+                currentPath = LabelUtils.EMPTY_PATH;
+                return Collections.emptyList();
+            } else if ("runComplete".equals(event)) {
+                List<TreeNodeEvent> closeAll = closeSuitePaths(currentPath, EMPTY_PATH)
+                        .map((Function<Path, TreeNodeEvent>) this::newTestSuiteFinishedEvent)
+                        .collect(Collectors.toList());
+                currentPath = LabelUtils.EMPTY_PATH;
+                return closeAll;
+            } else if (!"testCompleted".equals(event)) {
+                return Collections.emptyList();
+            }
+
+            Path path = toPath(obj);
+            if ("todo".equals(getStatus(obj))) {
+                path = path.resolve("todo");
+            }
+
+            List<TreeNodeEvent> result = Stream.of(
+                    closeSuitePaths(currentPath, path).map((Function<Path, TreeNodeEvent>) this::newTestSuiteFinishedEvent),
+                    openSuitePaths(currentPath, path).map((Function<Path, TreeNodeEvent>) this::newTestSuiteStartedEvent),
+                    testEvents(path, obj)
+            )
+                    .flatMap(Function.identity())
+                    .collect(Collectors.toList());
+
+            currentPath = path;
+            return result;
+
+        } catch (JsonSyntaxException e) {
+            if (text.contains("Compilation failed")) {
+                String json = text.substring(0, text.lastIndexOf("Compilation failed"));
+                JsonObject obj = gson.fromJson(json, JsonObject.class);
+                JsonElement type = obj.get("type");
```

Comment:
This line throws a NullPointerException when there's a compile error. I was able to reproduce it by trying to run the elm-test run config where one of my .elm files contains the following:

```elm
import Expect
import Test exposing (Test, describe, test)

suite : Test
suite =
    test "foo" ( \_ -> Expect.true )
```

I also noticed it happen for syntax errors.

Her'es the full stacktrace:
```
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor60.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.execution.process.ProcessHandler$5.invoke(ProcessHandler.java:223)
	at com.sun.proxy.$Proxy19.onTextAvailable(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:197)
	at com.intellij.execution.process.ColoredProcessHandler.textAvailable(ColoredProcessHandler.java:97)
	at com.intellij.execution.process.ColoredProcessHandler.coloredTextAvailable(ColoredProcessHandler.java:71)
	at com.intellij.execution.process.AnsiEscapeDecoder.processTextChunk(AnsiEscapeDecoder.java:247)
	at com.intellij.execution.process.AnsiEscapeDecoder.escapeText(AnsiEscapeDecoder.java:100)
	at com.intellij.execution.process.ColoredProcessHandler.notifyTextAvailable(ColoredProcessHandler.java:60)
	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:183)
	at com.intellij.util.io.BaseOutputReader.sendText(BaseOutputReader.java:211)
	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:195)
	at com.intellij.util.io.BaseOutputReader.readAvailableNonBlocking(BaseOutputReader.java:114)
	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:85)
	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:163)
	at com.intellij.util.io.BaseDataReader$1$1.run(BaseDataReader.java:66)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:229)
	at com.intellij.util.io.BaseDataReader$1.run(BaseDataReader.java:63)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at org.frawa.elmtest.core.ElmTestJsonProcessor.accept(ElmTestJsonProcessor.java:70)
	at org.frawa.elmtest.run.ElmTestRunProfileState$ConsoleProperties$1.processServiceMessages(ElmTestRunProfileState.java:114)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.processConsistentText(OutputToGeneralTestEventsConverter.java:104)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter$1.onLineAvailable(OutputToGeneralTestEventsConverter.java:54)
	at com.intellij.execution.testframework.sm.runner.OutputLineSplitter.onChunksAvailable(OutputLineSplitter.java:173)
	at com.intellij.execution.testframework.sm.runner.OutputLineSplitter.flushStdOutBuffer(OutputLineSplitter.java:150)
	at com.intellij.execution.testframework.sm.runner.OutputLineSplitter.processStdOutConsistently(OutputLineSplitter.java:105)
	at com.intellij.execution.testframework.sm.runner.OutputLineSplitter.processLine(OutputLineSplitter.java:70)
	at com.intellij.execution.testframework.sm.runner.OutputLineSplitter.process(OutputLineSplitter.java:53)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.process(OutputToGeneralTestEventsConverter.java:75)
	at com.intellij.execution.testframework.sm.SMTestRunnerConnectionUtil$2.onTextAvailable(SMTestRunnerConnectionUtil.java:208)
	... 25 more
```